### PR TITLE
[12.0][ADD] beesdoo_posorder_amount: by year

### PR DESCRIPTION
**Pull request by mclaeysb**
Using `groupby()`!

### CODE REVIEW

File: beesdoo_website_posorder_amount/templates/portal_posorder_amount.xml [hunk 1]
```diff
@@ -21,6 +21,24 @@
                           t-esc-options='{"widget": "monetary", "display_currency": "company_currency"}'/>
                     .
                 </p>
+                <table class="table table-striped">
```
Comment (robinkeunen) [inline beesdoo_website_posorder_amount/templates/portal_posorder_amount.xml]: Name your xml elements.

File: beesdoo_website_posorder_amount/controllers/main.py [hunk 2]
```diff
@@ -30,6 +32,15 @@ def _prepare_portal_layout_values(self):
         values["posorder_amount"] = sum(
             po.amount_total for po in owned_posorder
         )
+        values["posorder_amount_by_year"] = [
+            {
+                "year": key,
+                "amount": sum(element.amount_total for element in group),
+            }
+            for key, group in groupby(
+                owned_posorder, key=lambda element: element.date_order.year
```
Comment (robinkeunen) [inline beesdoo_website_posorder_amount/controllers/main.py]: better naming for better code ✊ 

```python
for key, group in groupby(
    owned_posorder, key=lambda pos_order: pos_order.date_order.year`
)
```

Comment (robinkeunen) [reply beesdoo_website_posorder_amount/controllers/main.py]: and even, to take Vincent's comment into account:

```py
        values["posorder_amount_by_year"] = [
            {
                "year": year,
                "amount": sum(pos_order.amount_total for pos_order in grouped_pos_orders),
            }
            for year, grouped_pos_orders in groupby(
                owned_posorder, key=lambda pos_order: pos_order.date_order.year
        )
    ]

```

**Review by robinkeunen — CHANGES_REQUESTED** [review]
Superb, glad to learn about groupby! I have strong opinions about naming though, as you know ;-)

**Comment by vvrossem** [comment]
ok to merge?

**Review by vvrossem — APPROVED** [review]

**Comment by vvrossem** [comment]
merging :) I see that Robin's requests were made

**Pull request merged**